WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/StartingInfo/ProductCreationAIStartingInfoViewModel.swift

```diff
@@ -50,20 +54,45 @@ final class ProductCreationAIStartingInfoViewModel: ObservableObject {
     }
 
     func didTapRemovePhoto() {
+        let previousState = imageState
```

Comment:
❓ Should we keep a reference of the previous state in the class's scope? I wonder if it can get deallocated when this method is finished, causing an undefined behavior when tapping the Undo button.

Reply:
Started capturing the `previousState` from inside the closure in https://github.com/woocommerce/woocommerce-ios/pull/13213/commits/26a747ab64e1059ce9008fdfb21e960bb30e2928